### PR TITLE
dev-lang/icon: EAPI 6 Cleanup

### PR DESCRIPTION
**Pull request by CheyenneWills**
Added comments concerning upstream requirement for emake -j1

Bug: https://bugs.gentoo.org/652178
Package-Manager: Portage-2.3.24, Repoman-2.3.6

### CODE REVIEW

File: dev-lang/icon/icon-9.5.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+src_compile() {
+	# Fails if more then one make job process.
+	# This is an upstream requirement.
+	emake -j1 CC="$(tc-getCC)" CFLAGS="${CFLAGS}"
+}
+
+src_test() {
+	make Samples || die "Samples failed"
```
Comment (Moiman) [inline dev-lang/icon/icon-9.5.1-r1.ebuild]: Can you use emake?

File: dev-lang/icon/icon-9.5.1-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit flag-o-matic toolchain-funcs
+
+MY_PV=${PV//./}
+SRC_URI="http://www.cs.arizona.edu/icon/ftp/packages/unix/icon-v${MY_PV}src.tgz"
+HOMEPAGE="http://www.cs.arizona.edu/icon/"
+DESCRIPTION="very high level language"
```
Comment (SoapGentoo) [inline dev-lang/icon/icon-9.5.1-r1.ebuild]: correct order:
```
DESCRIPTION
HOMEPAGE
SRC_URI
```

File: dev-lang/icon/icon-9.5.1-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+S="${WORKDIR}/icon-v${MY_PV}src"
+
+DEPEND="
+	X? (
+		x11-proto/xextproto
+		x11-proto/xproto
+		x11-libs/libX11
+		x11-libs/libXpm
+		x11-libs/libXt )
+	|| ( sys-devel/gcc sys-devel/gcc-apple )"
```
Comment (SoapGentoo) [inline dev-lang/icon/icon-9.5.1-r1.ebuild]: depending on compilers in `DEPEND` is pointless, remove

File: dev-lang/icon/icon-9.5.1-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+PATCHES=(
+	"${FILESDIR}/${PN}-9.5.1-flags.patch"
+)
+
+src_prepare() {
+	default
+
+	# do not prestrip files
+	find "${S}"/src -name 'Makefile' | xargs sed -i -e "/strip/d" || die
```
Comment (SoapGentoo) [inline dev-lang/icon/icon-9.5.1-r1.ebuild]: remove `"${S}"/`

File: dev-lang/icon/icon-9.5.1-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+	if use X; then
+		# Fails if more then one make job process.
+		# This is an upstream requirement.
+		emake X-Configure name=${mytarget} -j1 || die
+	else
+		# Fails if more then one make job process.
+		# This is an upstream requirement.
+		emake Configure name=${mytarget} -j1 || die
+	fi
```
Comment (SoapGentoo) [inline dev-lang/icon/icon-9.5.1-r1.ebuild]: change this whole block to `emake $(usex X X-Configure Configure) name=${mytarget} -j1 || die`

File: dev-lang/icon/icon-9.5.1-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+	emake -j1 CC="$(tc-getCC)" CFLAGS="${CFLAGS}"
+}
+
+src_test() {
+	make Samples || die "Samples failed"
+	make Test || die "Test failed"
+}
+
+src_install() {
+
```
Comment (SoapGentoo) [inline dev-lang/icon/icon-9.5.1-r1.ebuild]: remove empty line

File: dev-lang/icon/icon-9.5.1-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+	dosym ../$(get_libdir)/icon/bin/vib   /usr/bin/vib
+
+	cd "${S}/man/man1" || die
+	doman "${PN}"t.1
+	doman "${PN}".1
+	
+	cd "${S}/doc" || die
+	DOCS=( *.txt ../README )
+	
+	HTML_DOCS=( *.htm *.gif *.jpg *.css )
```
Comment (SoapGentoo) [inline dev-lang/icon/icon-9.5.1-r1.ebuild]: `HTML_DOCS=( *.{htm,gif,jpg,css} )`

**Comment by CheyenneWills** [comment]
Will make requested changes this evening

File: dev-lang/icon/icon-9.5.1-r1.ebuild (hunk 3 above)
Comment (CheyenneWills) [reply dev-lang/icon/icon-9.5.1-r1.ebuild]: Not sure if clang has been tested upstream, but I can remove the depend.

File: dev-lang/icon/icon-9.5.1-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+src_test() {
+	emake Samples || die "Samples failed"
+	emake Test || die "Test failed"
+}
+
+src_install() {
+	# Needed for make Install
+	dodir /usr/$(get_libdir)
+
+	emake Install dest="${D}/usr/$(get_libdir)/icon" || die "Make install failed"
```
[the remark this answers is not in the document]
Comment (CheyenneWills) [reply dev-lang/icon/icon-9.5.1-r1.ebuild]: Okay.. will remove the die on the emakes

**Pull request closed without merging**